File: codewars/Fun with tree - max sum/tree.py
from preloaded import TreeNode

def max_sum(root: TreeNode) -> int:
    total = 0
    if root is None:
        return total
    total += root.value
    r_root = max_sum(root.right)
    l_root = max_sum(root.left)
    if l_root == 0 and r_root != 0:
        return total + r_root
    elif r_root == 0 and l_root != 0:
        return total + l_root
    return total + max(l_root,r_root)

File: codewars/Fun with tree - max sum/tree.js
function maxSum(root) {
    if (root === null) {
      return 0; 
    }
    function dfs(node) {
      if (node === null) {
        return Number.NEGATIVE_INFINITY; 
      }
      if (node.left === null && node.right === null) {
        return node.value;
      }
      const leftSum = dfs(node.left);
      const rightSum = dfs(node.right);
      return node.value + Math.max(leftSum, rightSum);
    }
  
    return dfs(root);
  }